app/javascript/greetingsSlice.js

```diff
@@ -0,0 +1,39 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  greeting: '',
+  loading: false,
+  error: ''
+};
+
+export const fetchGreetings = createAsyncThunk('greetings/fetchGreetings', async() => {
+  try {
+    const response = await fetch('http://localhost:3000/api/v1/greetings');
```

Comment:
- [ ] Hello, when I run your application I get below error
![image](https://github.com/Akorede3133/react-rails-one-app/assets/112261237/0d761df8-7645-4a3b-8c79-a152bc0acaf1)

- When you use the full URL (http://localhost:3000/api/v1/greetings), it's treated as a cross-origin request because the protocol (http), domain (localhost), and port (3000) are different from the ones that served your React application. Browsers enforce the Same-Origin Policy for security reasons, and cross-origin requests are typically blocked unless the server explicitly allows them.

however when I change to relative path, request is successfully and get random greetings on the page
```suggestion
    const response = await fetch('/api/v1/greetings');
```

![image](https://github.com/Akorede3133/react-rails-one-app/assets/112261237/c23e4ba0-bb86-4539-ac0e-316f77ad125a)

- When you use a relative URL (/api/v1/greetings), the browser assumes that the request is intended for the same origin as the one that served the React application. In the context of a single-page application (SPA) served by a Rails backend, this kind of request is considered same-origin and is allowed by the browser.

Reply:
HI @zeff96, This is meant to be a react-rails app. I don't think they are meant to be run on separate ports. There is another project that requires the React app to be a separate one from the rails.
It's at that point we'll consider the CORS policy.
Kindly consider the above message.

Reply:
@Akorede3133, that is exactly what i was explaining, when you use full URL, browser treats that as a separate request thus blocks the request unless you use cors to allow such request, being you are serving react inside rails, just use relative path and the browser will treat as request from same origin and there wont be any error of cors